### PR TITLE
Test plans for https://issues.redhat.com/browse/QUARKUS-4179

### PR DESCRIPTION
**Pull request by fedinskiy**
### Links

JIRA: https://issues.redhat.com/browse/QUARKUS-4179

Quarkus documentation: https://quarkus.io/guides/security-openid-connect-client-reference#token-propagation-reactive

### Reminder for considerable topics

 - [x] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-4179.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Scope of the testing
+Support for quarkus-oidc-token-propagation and quarkus-oidc-token-propagation-reactive.
+Functionality, that should be supported:
+- Client accessing protected endpoint using token exchange grant
+- Usage of AccessTokenRequestFilter and JsonWebTokenRequestFilter
+- Classic and reactive
+
+### Existing tests
+- Test suite. Covers AccessTokenRequestFilter and usage of token request grant
```
Comment (rsvoboda) [inline QUARKUS-4179.md:16]: What testsuite - QE one?

File: QUARKUS-4179.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+## Scope of the testing
+Support for quarkus-oidc-token-propagation and quarkus-oidc-token-propagation-reactive.
+Functionality, that should be supported:
+- Client accessing protected endpoint using token exchange grant
+- Usage of AccessTokenRequestFilter and JsonWebTokenRequestFilter
+- Classic and reactive
+
+### Existing tests
+- Test suite. Covers AccessTokenRequestFilter and usage of token request grant 
+- Upstream
```
Comment (rsvoboda) [inline QUARKUS-4179.md:17]: integration tests or unit tests?

File: QUARKUS-4179.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- There is a relevant quickstart `security-openid-connect-client-quickstart`, part of quickstarts jenkins jobs since 2.7 
+
+### Automated test development
+- Test suite (there is some coverage already). 
+- Marete tests should check, that both of the extensions are present and are in tech preview status
```
Comment (rsvoboda) [inline QUARKUS-4179.md:22]:
```suggestion
- Marete tests must check, that both of the extensions are present and are in tech preview status
```

File: QUARKUS-4318.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+Ability to tie REST client with OIDC client for token propagation in filters.
+Functionality, that should be supported:
+- An ability for rest clients to have name of OIDC client as parameter. This client should be used for token exchange. 
+- Coverage in both classic and reactive mode.
+
+### Existing tests
+- Upstream covers that in https://github.com/quarkusio/quarkus/pull/39132 
+
+### Automated test development
+- Not required
```
Comment (rsvoboda) [inline QUARKUS-4318.md:18]: What about negative tests? Incorrect name reference, duplicated named definition, etc.

File: QUARKUS-4179.md (hunk 1 above)
Comment (fedinskiy) [reply QUARKUS-4179.md:16]: Yes

Comment (rsvoboda) [reply QUARKUS-4179.md:16]: ok, please be more concrete in the text

File: QUARKUS-4318.md (hunk 4 above)
Comment (rsvoboda) [reply QUARKUS-4318.md:18]: +1 to do some manual checks around this

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**